test/scripts/same_num_io.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# set -eu
+
+if [ ! -d $k4MarlinWrapper_tests_DIR/inputFiles/ ]; then
+  mkdir $k4MarlinWrapper_tests_DIR/inputFiles
+fi
+
+if [ ! -f $k4MarlinWrapper_tests_DIR/inputFiles/muons.slcio ]; then
+  wget https://github.com/AIDASoft/DD4hep/raw/master/DDTest/inputFiles/muons.slcio -P $k4MarlinWrapper_tests_DIR/inputFiles/
+fi
+
+../run k4run $k4MarlinWrapper_tests_DIR/gaudi_opts/same_num_io.py
```

Comment:
Could this be run with `--num-events` set to `-1` or a number greater than the input events, otherwise we are basically just testing the `EvtMax` handling of Gaudi itself, instead of testing that the k4MarlinWrapper handles EOF gracefully.

I think the original tests with `-1` and `15` as arguments to `--num-events` are exactly what we need, I would just put the check for the same number of events in the inputs and outputs to those tests, using a gaudi options file that also produces an `slcio` output file (I think the one that is used in those tests, doesn't produce an output(?)).